[JIT] Add more python ir utilities

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73329
* #73270
* __->__ #69871

Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

Commit:
Add python ir utilities

[ghstack-poisoned]

Commit:
Update on "Add python ir utilities"




[ghstack-poisoned]

eellison commented on torch/jit/_ir_utils.py:
```diff
@@ -0,0 +1,20 @@
+    def __exit__(self, *args):
+        self.g.setInsertPoint(self.prev_insert_point)
+
+def with_insert_point(self, insert_point: Union[torch._C.Node, torch._C.Block]):
```
Is `with g.with_insert_point` redundant? maybe `with g.insert_point_guard` or something else ? what do people think

ZolotukhinM replied:
I like `insert_point_guard` slightly better

Commit:
Update on "[JIT] Add more python ir utilities"




[ghstack-poisoned]

Commit:
Update on "[JIT] Add more python ir utilities"




[ghstack-poisoned]

Commit:
Update on "[JIT] Add more python ir utilities"




[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33515232).

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]

Commit:
Update on "[JIT] Add more python ir utilities"


Differential Revision: [D33515232](https://our.internmc.facebook.com/intern/diff/D33515232)

[ghstack-poisoned]
